src/components/Item/ItemModified.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { injectIntl } from 'react-intl';
+import messages from '../messages';
+import Datefield from '../Date';
+import { VIEW_RECENTS } from '../../constants';
+import type { BoxItem, View } from '../../flowTypes';
+
+type Props = {
+    item: BoxItem,
+    view: View,
+    intl: any
+};
+
+const ItemModified = ({ view, item, intl }: Props) => {
+    const { modified_at = '', interacted_at = '', modified_by }: BoxItem = item;
+    const isRecents: boolean = view === VIEW_RECENTS;
+    const date: string = isRecents ? interacted_at || modified_at : modified_at;
+    const modified_by_name: string = modified_by.name;
+    const byLabel = intl.formatMessage(messages.by);
+
+    return (
+   		<span className='be-item-label'>
+            <Datefield date={date} />&nbsp;{byLabel}&nbsp;{modified_by_name}
```

Comment:
Probably should not try to concatenate a line based on localization, because what maybe fine in one language, might be different in another. So doing `date by name` may not be written/spoken in that order in another locale.

Also, since its in-dom style, should use <FormattedMessage> in-line as opposed to intl, whenever possible.

So instead do it something like the rich text formatting example as shown here https://github.com/yahoo/react-intl/wiki/Components#formattedmessage

```
<FormattedMessage
    {...messages.modifiedColumnValue}
    values={{
        date: <Datefield date={date} />
        name: modifiedByName
    }}
/>
```

```
modifiedColumnValue: {
        id: 'be.modifiedColumnValue',
        description: 'Text for modified column',
        defaultMessage: '{date} by {name}:'
    },
```

Reply:
The other thing I will check, is that can the name be empty at times. Since UI is rendered based on auth token scope and does that get a name in all cases. If it doesn't we may have to drop the "by name" part for those cases or make it like `by Unknown` or something.